eth/tracers/internal/tracetest/util.go

```diff
@@ -47,6 +47,11 @@ func (c *callContext) toBlockContext(genesis *core.Genesis) vm.BlockContext {
 	if genesis.Config.IsLondon(context.BlockNumber) {
 		context.BaseFee = (*big.Int)(c.BaseFee)
 	}
+
+	if genesis.Config.TerminalTotalDifficulty != nil && genesis.Config.TerminalTotalDifficulty.Sign() == 0 {
```

Comment:
No, what I meant is that you should check the difficulty in the corresponding block header, rather than the `TerminalTotalDifficulty`

Reply:
you may misunderstand the func.

it's to get the block context from genesis config.

Reply:
Isn't it using the information in the genesis config to construct the block context for the call context?

Reply:
I think you need to read the context of the function, there is no header you mentioned

https://github.com/ethereum/go-ethereum/blob/a5fe7353cff959d6fcfcdd9593de19056edb9bdb/eth/tracers/internal/tracetest/calltrace_test.go#L89-L118